### PR TITLE
ablastr::particles::compute_weights : implement 1D and use template parameter to specify if field is nodal

### PR DESCRIPTION
**Pull request by lucafedeli88**
Currently WarpX does not compile on Frontier because the host function
`ABLASTR_ABORT_WITH_MESSAGE("Error: compute_weights not yet implemented in 1D");` is called inside the host device `ablastr::particles::compute_weights`. This PR fixes the issue by implementing `compute_weights` also for the 1D case.
In addition, an enum template parameter is used to tell the function if the field is nodal or cell-centered, instead of relying on an integer function parameter.

### CODE REVIEW

File: Source/ablastr/particles/NodalFieldGather.H [hunk 1]
```diff
@@ -17,11 +17,17 @@
 
 namespace ablastr::particles
 {
+
+    enum class FieldType{
+        nodal,
+        cell_centered
+    };
```
Comment (ax3l) [inline Source/ablastr/particles/NodalFieldGather.H]: Can we use `amrex::IndexType::NODE`/`amrex::IndexType::CELL` for this?

Comment (lucafedeli88) [reply Source/ablastr/particles/NodalFieldGather.H]: Good idea!

**Comment by lucafedeli88** [comment]
ping @ax3l, @RemiLehe ;-)

File: Source/ablastr/particles/NodalFieldGather.H [hunk 2]
```diff
@@ -30,24 +31,30 @@ namespace ablastr::particles
  * \param W        2D array of weights to store each neighbouring node (or cell-centered node)
  * \param nodal    Int that tells if the weights are calculated in respect to the nodes (nodal=1) of the cell-centered nodes (nodal=0)
  */
+template <amrex::IndexType::CellIndex IdxType>
```
Comment (ax3l) [inline Source/ablastr/particles/NodalFieldGather.H:36]: Should this file `#include <AMReX_IndexType.H>` for robustness now?

Comment (ax3l) [reply Source/ablastr/particles/NodalFieldGather.H:36]: Added :)

File: Source/ablastr/particles/NodalFieldGather.H [hunk 3]
```diff
@@ -12,16 +12,19 @@
...
+
 namespace ablastr::particles
 {
+
 /**
  * \brief Compute weight of each surrounding node (or cell-centered nodes) in interpolating a nodal ((or a cell-centered node) field
- *        to the given coordinates. If nodal=1, then the calculations will be done with respect to the nodes (default). If nodal=0, then the calculations will be done with respect to the cell-centered nodal)
- *
- * This currently only does linear order.
+ *        to the given coordinates. If template parameter IdxType is amrex::IndexType::CellIndex::NODE, then the calculations will be done with respect to the nodes (default).
```
Comment (ax3l) [inline Source/ablastr/particles/NodalFieldGather.H]: Is the documented default set?
```suggestion
 *        to the given coordinates. If template parameter IdxType is amrex::IndexType::CellIndex::NODE, then the calculations will be done with respect to the nodes.
```

**Review by ax3l — APPROVED** [review]
This looks great, thank you! :sparkles:

**Pull request merged**